src/API/Status/PreviewCard.vala

```diff
@@ -121,6 +129,47 @@ public class Tuba.API.PreviewCard : Entity, Widgetizable {
 		}
 	}
 
+	#if CLAPPER_0_10
+		private bool enhancer_check (string scheme, string? t_host = null) {
+			if (!Clapper.WITH_ENHANCERS_LOADER) return false;
+			bool supported = false;
+
+			string? host = t_host;
+			if (host != null && host.has_prefix ("www.")) host = host.substring (4);
+
+			var proxies = Clapper.get_global_enhancer_proxies ();
+			for (var i = 0; i < proxies.get_n_proxies (); i++) {
+				var proxy = proxies.get_proxy (i);
+
+				string? proxy_schemes = proxy.get_extra_data ("X-Schemes");
+				if (proxy_schemes == null || proxy_schemes == "") continue;
+
+				string[] schemes = proxy_schemes.split (";");
+				if (!(scheme in schemes)) continue;
```

Comment:
This part can probably be simplified to something like:

```vala
if (!proxy.extra_data_lists_value ("X-Schemes", scheme)) continue;
```

`proxy_schemes.split (";");` - I not a Vala expert, but I imagine this underneath does `malloc` to create char pointers array and `malloc` on each string in it, while Clapper provided `proxy.extra_data_lists_value` iterates it without any split/malloc so it might be a tiny bit faster.